typescript/packages/subsurface-viewer/src/layers/wells/wellsLayer.ts

```diff
@@ -299,9 +309,25 @@ export default class WellsLayer extends CompositeLayer<WellsLayerProps> {
             ...this.state,
             data,
             coarseData,
+            camChange: 0,
         });
     }
 
+    shouldUpdateState({ changeFlags }: UpdateParameters<this>): boolean {
+        if (changeFlags.viewportChanged) {
+            this.setState({
+                ...this.state,
+                camChange: (this.state["camChange"] as number) + 1,
```

Comment:
Should this be updated in `updateState()` instead? YCR.

Reply:
I belive it is best where it is as one has to pick up camerachange even if not all state is updated